docs/extend/modular-resources/create-type.md

```diff
@@ -0,0 +1,64 @@
+---
+title: "Define a New Resource Type"
+linkTitle: "New API Type"
+weight: 30
+type: "docs"
+tags: ["rdk", "extending viam", "modular resources", "API"]
+description: "Define an API for a new type of resource."
+no_list: true
+---
+
+You need to define a new {{% glossary_tooltip term_id="resource" text="resource" %}} *type* (and an API for that type) if:
+
+- You have a {{% glossary_tooltip term_id="resource" text="resource" %}} that does not fit into any of the existing [component](/components/) or [service](/services/) types.
+- You have a resource that could fit into an existing type, for example, [base](/components/base/), but you want to define an API with different endpoints than the ones in the standard Viam [API](/program/apis/) for that type.
+
+{{% alert title="Tip" color="tip" %}}
+
+If you want to use most of an existing API but need just a few other functions, try using the [`DoCommand`](/program/apis/#docommand) endpoint to add custom functionality to an existing type.
+
+{{% /alert %}}
+
+## How to define your new resource API
+
+Viam uses [protocol buffers](https://protobuf.dev/) for API definition.
+
+To define a new type, you need to define the messages and methods of the new API in [protobuf](https://github.com/protocolbuffers/protobuf), generate all necessary [protobuf module files](https://buf.build/docs/generate/usage/), and then write code in Python or Go to implement the higher level server and client functions required.
+The following steps guide you through this process in more detail.
+
+1. Decide whether your type is a component or a service.
+  If it is hardware, it is a component.
+  If it is a software package, it is a service.
```

Comment:
```suggestion
  If it is providing higher-level functionality, it is a service.
```

Since all of it is really software, and some services are still interacting with the underlying hardware (for example, motion or vision)

Reply:
```suggestion
  If it provides higher-level functionality, it is a service.
```